app/components/review_summary/view.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module ReviewSummary
+  class View < GovukComponent::Base
+    renders_one :header
+
+    def initialize(form:, trainee:, data_view: ApplyInvalidDataView.new(trainee.apply_application))
+      @has_errors = form.respond_to?(:errors) ? form.errors.any? : false
+      @form = form
+      @trainee = trainee
+      @invalid_data_view = data_view
+    end
+
+    def render?
+      errors_captured? || invalid_data_exists?
+    end
+
+    def summary_component
+      @summary_component ||= errors_captured? ? ErrorSummary::View : InformationSummary::View
```

Comment:
You could put this in the `initialize` method and create a public accessor method. Saves you from having to create a separate method and then having to memoise the result.